app/controllers/images_controller.rb

```diff
@@ -244,4 +254,26 @@ def update_crop_params
     # FIXME: this will raise a warning because of unpermitted paramaters
     params.permit(:crop_x, :crop_y, :crop_width).merge(crop_height: crop_height.to_i)
   end
+
+  def next_lead_image_revision(revision, image_revision, selected)
+    if revision.lead_image_revision&.image_id == image_revision.image_id
+      return nil unless selected
+    end
+
+    if revision.lead_image_revision&.image_id != image_revision.image_id
+      return image_revision if selected
+    end
+
+    revision.lead_image_revision
+  end
```

Comment:
I found this method a bit tricky to read

is it along the lines of:

```
return image_revision if selected

currently_lead = revision.lead_image_revision&.image_id == image_revision.image_id
if currently_lead && !selected
  nil
else
  revision.lead_image_revision
end
```

Or am I missing something?

Reply:
That's much better :-). Originally, we also had to determine if there had been a change, and I didn't see the refactoring opportunity now that logic is separated.